### ISSUE TITLE
gtk-doc test failure

### ISSUE THREAD

**Issue by markhindley** [ConsoleKit2/ConsoleKit2#130]
First, many thanks for reviving consolekit2. It is great to see activity here again.

I have packaged version 1.2.4 for Devuan.

During that, I came across a test failure when building the gtk-docs:

> Running suite(s): gtk-doc-libconsolekit
> /build/consolekit2-1.2.4/doc/libconsolekit/libconsolekit-docs.xml:1:E: doesn't appear to include "xml/api-index-1.2.4.xml"
> 75.0%: Checks 4, Failures: 1
> FAIL gtkdoc-check.test (exit status: 1)

I am unsure where the xml/api-index-1.2.4.xml comes from. xml/api-index-1.0.xml also exists and is correctly included in the generated docs.

Thanks.

Mark

**Comment by ajacoutot** [comment]
Hey @markhindley 
Thanks for the report, could you provide an easy way to reproduce this?
Thanks.

**Comment by markhindley** [comment]
On Wed, Nov 30, 2022 at 11:03:54AM +0000, Antoine Jacoutot wrote:
>    Hey @markhindley
>    Thanks for the report, could you provide an easy way to reproduce this?

The Devuan build is configured with

 ./configure --enable-libudev \
	--enable-udev-acl \
	--enable-inotify
	--libexecdir="\$$(prefix)/lib/ConsoleKit" \
	--with-rundir=/run \
	--with-xinitrc-dir=/etc/X11/Xsession.d \
	--enable-polkit \
	--enable-libevdev \
	--enable-libdrm \
	--enable-libselinux \
	--enable-pam-module \
	--with-pid-file=/run/console-kit-daemon.pid \
	--without-systemdsystemunitdir \
	--enable-docbook-docs \
	--enable-gtk-doc \
	--with-pam-module-dir=/lib/$(DEB_HOST_MULTIARCH)/security

Then make..

Mark

**Comment by ajacoutot** [comment]
Hey. I just tried to reproduce this but couldn't.
Are you running autogen.sh with gtk-doc installed before doing configure?

**Comment by markhindley** [comment]
I have just tried again  and ensured autogen.sh is run with the same result.

Log attached

Mark
[consolekit.gtk-doc.failure.txt](https://github.com/ConsoleKit2/ConsoleKit2/files/10898162/consolekit.gtk-doc.failure.txt)

**Comment by ajacoutot** [comment]
Hey @markhindley 
Thanks for you log.
It's actually failing when doing "make check", not just "make".
Could you try this.

```
diff --git a/doc/libconsolekit/libconsolekit-docs.xml b/doc/libconsolekit/libconsolekit-docs.xml
index 6f876b7..3ff4c4f 100644
--- a/doc/libconsolekit/libconsolekit-docs.xml
+++ b/doc/libconsolekit/libconsolekit-docs.xml
@@ -38,6 +38,11 @@
     <xi:include href="xml/api-index-1.0.xml"><xi:fallback /></xi:include>
   </index>
 
+  <index id="api-index-1.2">
+    <title>Version 1.2 API Index</title>
+    <xi:include href="xml/api-index-1.2.4.xml"><xi:fallback /></xi:include>
+  </index>
+
   <index id="api-index-deprecated">
     <title>Deprecated API Symbols</title>
     <xi:include href="xml/api-index-deprecated.xml"><xi:fallback /></xi:include>
```

**Comment by markhindley** [comment]
Antonie,
On Tue, Mar 07, 2023 at 09:17:20AM +0000, Antoine Jacoutot wrote:
>    Hey @markhindley
>    Thanks for you log.
>    It's actually failing when doing "make check", not just "make".
>    Could you try this.

Yes, that fixes it for me. Many thanks.

Mark

**Comment by ajacoutot** [comment]
FIxed with https://github.com/ConsoleKit2/ConsoleKit2/commit/984ac8c2c0814931b4e1e60fafe1c0420d1e5844

**Comment by markhindley** [comment]
On Tue, Mar 07, 2023 at 01:48:58PM +0000, Antoine Jacoutot wrote:
>    FIxed with 984ac8c

Unfortunately, the fix appears broken with the new release.

1.2.5 make check is failing for me with:

Making check in libconsolekit
make[3]: Entering directory '/build/consolekit2-1.2.5/doc/libconsolekit'
make  check-TESTS
make[4]: Entering directory '/build/consolekit2-1.2.5/doc/libconsolekit'
make[5]: Entering directory '/build/consolekit2-1.2.5/doc/libconsolekit'
echo "#!/bin/sh -e" > gtkdoc-check.test; \
	echo "/usr/bin/gtkdoc-check || exit 1" >> gtkdoc-check.test; \
	chmod +x gtkdoc-check.test
FAIL: gtkdoc-check.test
=========================================================
   ConsoleKit2 1.2.5: doc/libconsolekit/test-suite.log
=========================================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: gtkdoc-check
==================

Running suite(s): gtk-doc-libconsolekit
libconsolekit-undocumented.txt:1:E: 2 undocumented or incomplete symbols
libconsolekit-unused.txt:1:E: 16 unused documentation entries

50.0%: Checks 4, Failures: 2
FAIL gtkdoc-check.test (exit status: 1)

============================================================================
Testsuite summary for ConsoleKit2 1.2.5
============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See doc/libconsolekit/test-suite.log
Please report to https://github.com/ConsoleKit2/ConsoleKit2/issues
============================================================================

**Comment by markhindley** [comment]
Actually, now I look more closely this is a new issue: see #133.

Mark
